extension/ts/extensions/warning.ts

```diff
@@ -22,13 +13,13 @@ const addUndoEditListener = () =>
 		document.getElementById("book_editTabTextDelete"), // so that it doesn't alert you when you're deleting something (?)
 	].forEach((element) => element?.addEventListener("click", undoEdits));
 
-window.addEventListener(FORM_RENDER_EVENT, () => {
-	addEditListener();
+onFormRender(() => {
+	undoEdits();
 	addUndoEditListener();
 });
 
 window.addEventListener("beforeunload", (event) => {
-	if (edited) {
```

Comment:
This was FASTER.

At one point, I checked both, but it was only faster in the case where no one made any changes to any fields. So it didn't seem worth it to add the code complexity to make it literally 2 milliseconds faster on occasion.